### ISSUE TITLE
Different flags for C and C++

### ISSUE THREAD

**Issue by uranusjr** [nirm03/SublimeLinter-clang#8]
Currently it doesn’t seem possible to specify custom flags for C or C++-only. I usually set `-std=c++11` for C++, but this does not work when compiling C, and adding it to either `args` or `extra_flags` causes the linter to stop working altogether.

**Comment by GhostLyrics** [comment]
I added support for exactly this situation in pull request #10.

**Comment by nirm03** [comment]
Sorry for the late  response. Doesn't it suffice to simply use the .sublimelinerrc files?
Do you have C and C++ files in the same directory?

**Comment by GhostLyrics** [comment]
I am not sure. If I understood the idea of the linterrc files correctly I would need to litter almost every directory of a university project (https://github.com/IAIK/sweb "operating systems" course) with linterrc files. The way I do it now - without be certain that it's the best way - is to use the following options in SublimeLinter.sublime-settings.

``` json
"clang": {
                "@disable": false,
                "args": [],
                "excludes": [],
                "extra_flags_all": "",
                "extra_flags_c": "-std=c11",
                "extra_flags_cxx": "-std=c++11",
                "include_dirs": [
                    "/Users/ghostlyrics/Repositories/sweb/arch/include",
                    "/Users/ghostlyrics/Repositories/sweb/arch/common/include",
                    "/Users/ghostlyrics/Repositories/sweb/arch/x86/32/common/include",
                    "/Users/ghostlyrics/Repositories/sweb/arch/x86/32/include",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/console",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/fs",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/fs/devicefs",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/fs/minixfs",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/fs/ramfs",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/kernel",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/mm",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/ustl",
                    "/Users/ghostlyrics/Repositories/sweb/common/include/util",
                    "/Users/ghostlyrics/Repositories/sweb/userspace/libc/include/sys",
                    "/Users/ghostlyrics/Repositories/sweb/userspace/libc/include"
                ]
            }
```

If you know of a better way I'm happy to learn. (That is, I'm not using sublime project files because they always caused Sublime to edit the project file instead of loading the "workspace" with all the files and settings.)

**Comment by nirm03** [comment]
sublimelinterrc files affect whole subtrees including all of their subdirectories, so if for example, you project roughly looks like this:
cpp_subtree
c_subtree
then two sublimelinterrc files should do the trick.

If however, your project truly contains C/C++ files scattered randomly, then I agree that sublimelinterrc files are not a valid solution.

From my experience though, the latter case is quite uncommon, and somewhat undesirable anyway,
but of course, your mileage may vary.
Are you saying this latter case is a common scenario for you?

**Comment by GhostLyrics** [comment]
Short version: No, it is not a common scenario.

Longer version:
For this project only the files in `…/userspace/…` are C, the rest should be C++. I threw up this pull request because I thought it might be helpful to be able to define these settings and it was relatively easy work. I tend to not work with C and C++ mixed projects at all, apart from this course.
Most of the university or work projects I work at are either completely in C or completely in C++. (Or a completely different programming language for that matter.)

**Comment by nirm03** [comment]
I see. So if we take your project as an example, couldn't you have just done the following:
1. Put a sublimelinterrc file in the top of the project tree with C++ syntax.
2. Put a sublimelinterrc file in the userspace directory with C syntax.
This way you're getting C syntax for the userspace directory, and C++ for the rest.
Would that satisfy your request, or am I missing something?

**Comment by GhostLyrics** [comment]
Apart from the error that it does not work with C only if you give a C only flag, yes. I corrected that in the lower part of ee135a3, which seems to be what is being discussed in #14. If you want me to, I can split that out and send a new PR for the C/C++ separation part.

(Sorry for the delayed reply. I had trouble testing your suggestions and was tempted to accuse your code of being bad but realized that the default `rc_search_limit` was too little for this project :blush:)
